lib/features/websocket/application/socket_cubit.dart

```diff
@@ -0,0 +1,129 @@
+import 'dart:convert';
+
+import 'package:asset_tracker/features/websocket/domain/currency.dart';
+import 'package:asset_tracker/features/websocket/domain/meta.dart';
+import 'package:asset_tracker/features/websocket/domain/price_changed_data_model.dart';
+import 'package:asset_tracker/features/websocket/infrastructure/socket_repository.dart';
+import 'package:bloc/bloc.dart';
+import 'dart:async';
+
+part 'socket_state.dart';
+
+class SocketCubit extends Cubit<SocketState> {
+  final SocketRepository socketRepository;
+  PriceChangedDataModel currentData =
+      PriceChangedDataModel(data: {}, meta: Meta(time: 0));
+
+  SocketCubit({required this.socketRepository}) : super(SocketInitial()) {
+    _connect();
+  }
+
+  void _connect() {
+    emit(SocketLoading());
+    try {
+      socketRepository.connect();
+      emit(SocketConnected());
+      _listenToSocket();
+    } catch (e) {
+      emit(SocketError(e.toString()));
+      _retryConnection();
+    }
+  }
+
+  void _listenToSocket() {
+    socketRepository.channel.stream.listen(
+      (data) {
+        if (data.toString().startsWith('0')) {
+          print('Data 0 ile aslıyor 40 gönnder');
+          sendMessage('40');
+          emit(SocketLoading());
+        } else if (data.toString().startsWith('42')) {
+          if (currentData.data.isEmpty) {
+            print('Data set for the first time');
+            PriceChangedDataModel firstData = parseReceivedData(
+                data.toString().substring(19, data.toString().length - 1));
+
+            currentData = firstData;
+            emit(SocketDataReceived(currentData));
+          } else {
+            print('Data updated');
+            updatedData(parseReceivedData(
+                data.toString().substring(19, data.toString().length - 1)));
+            print(currentData.data);
+            emit(SocketDataReceived(currentData));
+          }
+        }
+      },
+      onError: (error) {
+        emit(SocketError(error.toString()));
+        _retryConnection();
+      },
+      onDone: () {
+        emit(SocketDisconnected(currentData));
+        _retryConnection();
```

Comment:
onDone kısmı sana sorun çıkartabilir. Sürekli veri çekmek istediğimiz için emit SocketDisconnect yapmamıza ihtiyaç yok. Ayrıca connection'u kapatmadık, retry demek yerine 40 yollarsak veriler yeniden gelecektir.